dspace-api/src/main/java/org/dspace/health/Core.java

```diff
@@ -0,0 +1,185 @@
+        long total_size = 0;
+
+        Collections.sort(col_bitSizes, new Comparator<Object[]>() {
+            @Override
+            public int compare(Object[] o1, Object[] o2) {
+                try {
+                    return CollectionDropDown.collectionPath((Collection) o1[0]).compareTo(
+                        CollectionDropDown.collectionPath((Collection) o2[0])
+                    );
+                } catch (Exception e) {
+                    ret.append(e.getMessage());
+                }
+                return 0;
+            }
+        });
+        for (Object[] row : col_bitSizes) {
+            Long size = (Long) row[1];
+            total_size += size;
+            Collection col = (Collection) row[0];
+            ret.append(String.format(
+                    "\t%s:  %s\n", CollectionDropDown.collectionPath(col), FileUtils.byteCountToDisplaySize((long) size)));
+        }
+        ret.append(String.format(
+                "Total size:              %s\n", FileUtils.byteCountToDisplaySize(total_size)));
+
+        ret.append(String.format(
+                "Resource without policy: %d\n", getBitstreamsWithoutPolicyCount()));
+
+        ret.append(String.format(
+                "Deleted bitstreams:      %d\n", getBitstreamsDeletedCount()));
+
+        String list_str = "";
+        List<UUID> bitstreamOrphans = getBitstreamOrphansRows();
+        for (UUID id : bitstreamOrphans) {
+            list_str += String.format("%d, ", id);
+        }
+        ret.append(String.format(
+                "Orphan bitstreams:       %d [%s]\n", bitstreamOrphans.size(), list_str));
+
+        return ret.toString();
+    }
+
+    public  String getObjectSizesInfo() throws SQLException {
+        String ret = "";
+        for (String tb : new String[] { "Bitstream", "Bundle", "Collection",
+            "Community", "MetadataValue", "EPerson", "Item", "Handle",
+            "Group", "BasicWorkflowItem", "WorkspaceItem", }) {
+            int count = count("SELECT COUNT(*) from " + tb);
+            ret += String.format("Count %-14s: %s\n", tb,
+                String.valueOf(count));
+        }
+        return ret;
+    }
+
+
+    // get objects
+    //
+
+    public  List<Object[]> getWorkspaceItemsRows() throws SQLException {
+        return query("SELECT wi.stageReached, count(*) as cnt from WorkspaceItem wi group by wi.stageReached order by wi.stageReached");
+    }
+
+    public  List<UUID> getBitstreamOrphansRows() throws SQLException {
+        return query("select bit.id from Bitstream bit where bit.deleted != true" +
+                " and bit.id not in (select bit2.id from Bundle bun join bun.bitstreams bit2)" +
+                " and bit.id not in (select com.logo.id from Community com)" +
+                " and bit.id not in (select col.logo.id from Collection col)" +
+                " and bit.id not in (select bun.primaryBitstream.id from Bundle bun)");
+    }
```

Comment:
@tdonohue 
I went with the "query for query" approach to keep the amount of changes to the PR at minimum (didn't work out as expected)

I still didn't get my head around the services api and am looking for the methods in the wrong places...

Few questions though - what's the deal with Lists and Iterators? I mean a half of BitstreamDAO methods returns a List the other an Iterator for no obvious reason...

For the various count() methods (that are not already implemented) is it preferred to create new queries in the DAO or use "what's there". As you've pointed for getBitstreamDeletedCount() you can call findDeletedBitstream(...).size or I could copy over the hql query I have.

And how would you go with the aggregate queries? `getWorkspaceItemsRows` actually returns counts grouped by stageReached (+ the stageReached "id").
I don't see this kind of method/query fitting in the api. What would fit me there is some kind of `findAllFilteredByStageReached` and then having a `foreach stageReached {findAllFilteredByStage(stageReached).size...}` sort of loop, or you know `countByStageReached`. Both seem equally stupid when I know I'll be needing the counts for all the stages...maybe that's optimizing prematurely, or maybe I'm missing a point somewhere...